pkg/service/service.go

```diff
@@ -853,22 +910,61 @@ func CreateServiceFromKubernetesInlineComponents(client *kclient.Client, k8sComp
 		// add labels to the CRD before creation
 		d.AddComponentLabelsToCRD(labels)
 
+		crdName, ok := getCRDName(d.OriginalCRD)
+		if !ok {
+			continue
+		}
+
+		delete(deployed, cr+"/"+crdName)
+
 		// create the service on cluster
 		err = client.CreateDynamicResource(d.OriginalCRD, group, version, resource)
 		if err != nil {
 			if strings.Contains(err.Error(), "already exists") {
 				// this could be the case when "odo push" was executed after making change to code but there was no change to the service itself
 				// TODO: better way to handle this might be introduced by https://github.com/openshift/odo/issues/4553
-				err = nil
 				continue // this ensures that services slice is not updated
 			} else {
-				return []string{}, err
+				return err
 			}
 		}
 
 		name, _ := d.GetServiceNameFromCRD() // ignoring error because invalid yaml won't be inserted into devfile through odo
-		services = append(services, strings.Join([]string{kind, name}, "/"))
+		created = append(created, strings.Join([]string{kind, name}, "/"))
 	}
 
-	return services, nil
+	for key := range deployed {
+		err = DeleteOperatorService(client, key)
+		if err != nil {
+			return err
+
+		}
+		deleted = append(deleted, key)
+	}
+
+	if len(created) == 1 {
+		log.Infof("Created service %q on the cluster; refer %q to know how to link it to the component", created[0], "odo link -h")
+	} else if len(created) > 1 {
+		log.Infof("Created services %q on the cluster; refer %q to know how to link them to the component", strings.Join(created, ", "), "odo link -h")
+	}
+
+	if len(deleted) == 1 {
+		log.Infof("Deleted service %q from the cluster", deleted[0])
+	} else if len(deleted) > 1 {
+		log.Infof("Deleted services %q from the cluster", strings.Join(deleted, ", "))
```

Comment:
Since `Creating Kubernetes resources....` also uses `log.Infof`, it might be better idea to use `log.Successf`
```
Creating Kubernetes resources for component nodejs-project-fexd
Created service "AtlasProject/my-project" on the cluster; refer "odo link -h" to know how to link it to the component
 ✓  Waiting for component to start [2s]
 ✓  Waiting for component to start [328ms]
```
`Successf` output
```
Creating Kubernetes resources for component nodejs-nodejs-basic-thqi
 ✓  Created service "AtlasCluster/my-atlas-cluster" on the cluster; refer "odo link -h" to know how to link it to the component
 ✓  Waiting for component to start [3s]
 ✓  Waiting for component to start [255ms]
```